nullaway/src/main/java/com/uber/nullaway/NullabilityUtil.java

```diff
@@ -307,7 +307,12 @@ public static TreePath findEnclosingMethodOrLambdaOrInitializer(TreePath path) {
    */
   public static boolean mayBeNullFieldFromType(@Nullable Symbol symbol, Config config) {
     if (symbol == null) {
-      return true;
+      // An expression representing a field dereference should always have a non-null symbol.  A
+      // `MemberSelectTree` _can_ have a null symbol in cases where it does _not_ represent a field
+      // dereference.  E.g., in a `requires` clause in a module-info.java file, the fully-qualified
+      // package name is represented using `MemberSelectTree`s.  When the symbol is null, there is
+      // no possibility of a null dereference.
+      return false;
```

Comment:
Is this the best place to filter these out? Or could we somehow check, before we call `mayBeNullFieldFromType(...)` if we truly have a field vs requires or any other `MemberSelectTree` using construct that isn't a method call or field dereference?

Reply:
Yeah, I wish we could check better to enumerate the cases where we expect the symbol may be null.  `RequiresTree` was introduced in JDK 9, but with some slightly creative code and the `TreePath` to a `MemberSelectTree` we could probably detect that case.

That said, I do not see how it is possible to have a valid field dereference or method call `MemberSelectTree` that does not have a symbol.  If there is no symbol, that means the compiler could not resolve the field / method, and the code simply shouldn't compile, unless I am missing something.  We could add some stronger assertions around that.

My only hesitation is that adding stronger assertions might cause another regression / crash where things were working before.  But I guess that's better than not understanding what is going on 🙂 Let me tinker a bit.